tests/components/rtsp_to_webrtc/test_init.py

```diff
@@ -143,21 +128,11 @@ async def test_offer_failure(
         exc=aiohttp.ClientError,
     )
 
-    client = await hass_ws_client(hass)
-    await client.send_json(
-        {
-            "id": 2,
-            "type": "camera/web_rtc_offer",
-            "entity_id": "camera.demo_camera",
-            "offer": OFFER_SDP,
-        }
-    )
-    response = await client.receive_json()
-    assert response.get("id") == 2
-    assert response.get("type") == TYPE_RESULT
-    assert "success" in response
-    assert not response.get("success")
-    assert "error" in response
-    assert response["error"].get("code") == "web_rtc_offer_failed"
-    assert "message" in response["error"]
-    assert "RTSPtoWebRTC server communication failure" in response["error"]["message"]
+    camera = get_camera_from_entity_id(hass, "camera.demo_camera")
```

Comment:
Offline we talked about how i was under the impression that the design pattern is that integration details should not be interested with in tests https://developers.home-assistant.io/docs/development_testing?_highlight=tests#writing-tests-for-integrations
and that higher level APIs should be used (e.g. the websocket API, or a model level function). You said you'll get a second opinion on this.

Just leaving a note here.